### PR TITLE
[Guidance Cards][Part 2]Defer to UIAppearance proxy for color/font literals

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
Builds on #2401 

This PR changes the UI logic in the `InstructionCardContainerView` and `InstructionCardView` to defer to the color/font literals supplied by the UIAppearance proxy.

### CODE REVIEW

File: MapboxNavigation/InstructionsCardContainerView.swift [hunk 1]
```diff
@@ -129,18 +129,17 @@ public class InstructionsCardContainerView: StylableView {
         setGradientLayer(for: lanesView)
         setGradientLayer(for: nextBannerView)
         
-        layer.cornerRadius = style.cornerRadius
+        layer.cornerRadius = 20.0
```
Comment (avi-c) [inline MapboxNavigation/InstructionsCardContainerView.swift:132]: This property should still be overridable through the new mechanism. We use it in Apex. The mobile (iPhone) UI uses a different corner radius than the in-car (iPad) UI.

File: MapboxNavigation/InstructionsCardView.swift [hunk 2]
```diff
@@ -52,26 +52,19 @@ public class InstructionsCardView: BaseInstructionsBannerView {
...
-        distanceLabel.valueTextColor = style.distanceLabelValueTextColor
-        distanceLabel.unitTextColor = style.distanceLabelUnitTextColor
-        
-        distanceLabel.valueFont = style.distanceLabelValueFont
-        distanceLabel.unitFont = style.distanceLabelUnitFont
+        distanceLabel.showHighlightedTextColor = false
     }
     
     fileprivate func preparePrimaryLabel(_ style: InstructionsCardStyle) {
-        primaryLabel.normalFont = style.primaryLabelNormalFont
```
Comment (1ec5) [inline MapboxNavigation/InstructionsCardView.swift:68]: @avi-c, is there still a need to customize the font used in the instruction card’s labels?

**Comment by 1ec5** [comment]
Some of the guidance card snapshot tests are failing. It might just be that the test fixtures need to be updated:

```
Test Case '-[MapboxNavigationTests.GuidanceCardsSnapshotTests testLanesManeuver]' started.
2020-08-03 20:35:47.494339+0000 Example[84993:147072] Task <3B98A0C3-990E-4CB4-83CE-CC40BAD1AC05>.<1> load failed with error Error Domain=NSURLErrorDomain Code=-999 "cancelled" UserInfo={NSErrorFailingURLStringKey=https://s3.amazonaws.com/mapbox/shields/v3/i-25@3x.png, NSErrorFailingURLKey=https://s3.amazonaws.com/mapbox/shields/v3/i-25@3x.png, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <3B98A0C3-990E-4CB4-83CE-CC40BAD1AC05>.<1>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <3B98A0C3-990E-4CB4-83CE-CC40BAD1AC05>.<1>, NSLocalizedDescription=cancelled} [-999]
2020-08-03 20:35:47.634902+0000 Example[84993:146893] If you have Kaleidoscope installed you can run this command to see an image diff:
ksdiff "/Users/distiller/project/MapboxNavigationTests/FailureDiffs/MapboxNavigationTests.GuidanceCardsSnapshotTests/reference_testLanesManeuver_iPhone6Plus_Portrait_iOS_12.2@3x.png" "/Users/distiller/project/MapboxNavigationTests/FailureDiffs/MapboxNavigationTests.GuidanceCardsSnapshotTests/failed_testLanesManeuver_iPhone6Plus_Portrait_iOS_12.2@3x.png"
/Users/distiller/project/MapboxNavigationTests/GuidanceCardsSnapshotTests.swift:63: error: -[MapboxNavigationTests.GuidanceCardsSnapshotTests testLanesManeuver] : failed - Snapshot comparison failed: Optional(Error Domain=FBSnapshotTestControllerErrorDomain Code=4 "Images different" UserInfo={NSLocalizedFailureReason=image pixels differed by more than 0.00% from the reference image, FBDiffedImageKey=<UIImage: 0x6000024773a0>, {414, 736}, FBReferenceImageKey=<UIImage: 0x60000247b170>, {414, 736}, FBCapturedImageKey=<UIImage: 0x600002477020>, {414, 736}, NSLocalizedDescription=Images different})
Test Case '-[MapboxNavigationTests.GuidanceCardsSnapshotTests testLanesManeuver]' failed (0.487 seconds).
Test Case '-[MapboxNavigationTests.GuidanceCardsSnapshotTests testRegularManeuver]' started.
2020-08-03 20:35:48.052319+0000 Example[84993:146893] If you have Kaleidoscope installed you can run this command to see an image diff:
ksdiff "/Users/distiller/project/MapboxNavigationTests/FailureDiffs/MapboxNavigationTests.GuidanceCardsSnapshotTests/reference_testRegularManeuver_iPhone6Plus_Portrait_iOS_12.2@3x.png" "/Users/distiller/project/MapboxNavigationTests/FailureDiffs/MapboxNavigationTests.GuidanceCardsSnapshotTests/failed_testRegularManeuver_iPhone6Plus_Portrait_iOS_12.2@3x.png"
/Users/distiller/project/MapboxNavigationTests/GuidanceCardsSnapshotTests.swift:40: error: -[MapboxNavigationTests.GuidanceCardsSnapshotTests testRegularManeuver] : failed - Snapshot comparison failed: Optional(Error Domain=FBSnapshotTestControllerErrorDomain Code=4 "Images different" UserInfo={NSLocalizedFailureReason=image pixels differed by more than 0.00% from the reference image, FBDiffedImageKey=<UIImage: 0x600002470d90>, {414, 736}, FBReferenceImageKey=<UIImage: 0x600002466140>, {414, 736}, FBCapturedImageKey=<UIImage: 0x60000247b020>, {414, 736}, NSLocalizedDescription=Images different})
Test Case '-[MapboxNavigationTests.GuidanceCardsSnapshotTests testRegularManeuver]' failed (0.416 seconds).
Test Case '-[MapboxNavigationTests.GuidanceCardsSnapshotTests testTertiaryManeuver]' started.
2020-08-03 20:35:48.470854+0000 Example[84993:146893] If you have Kaleidoscope installed you can run this command to see an image diff:
ksdiff "/Users/distiller/project/MapboxNavigationTests/FailureDiffs/MapboxNavigationTests.GuidanceCardsSnapshotTests/reference_testTertiaryManeuver_iPhone6Plus_Portrait_iOS_12.2@3x.png" "/Users/distiller/project/MapboxNavigationTests/FailureDiffs/MapboxNavigationTests.GuidanceCardsSnapshotTests/failed_testTertiaryManeuver_iPhone6Plus_Portrait_iOS_12.2@3x.png"
/Users/distiller/project/MapboxNavigationTests/GuidanceCardsSnapshotTests.swift:87: error: -[MapboxNavigationTests.GuidanceCardsSnapshotTests testTertiaryManeuver] : failed - Snapshot comparison failed: Optional(Error Domain=FBSnapshotTestControllerErrorDomain Code=4 "Images different" UserInfo={NSLocalizedFailureReason=image pixels differed by more than 0.00% from the reference image, FBDiffedImageKey=<UIImage: 0x60000247ff00>, {414, 736}, FBReferenceImageKey=<UIImage: 0x6000024778e0>, {414, 736}, FBCapturedImageKey=<UIImage: 0x600002465a40>, {414, 736}, NSLocalizedDescription=Images different})
Test Case '-[MapboxNavigationTests.GuidanceCardsSnapshotTests testTertiaryManeuver]' failed (0.418 seconds).
```

File: MapboxNavigation/InstructionsCardView.swift (hunk 2 above)
Comment (avi-c) [reply MapboxNavigation/InstructionsCardView.swift:68]: Yes, itt should still be supported if we want this to be truly flexible.

**Comment by avi-c** [comment]
The test failures are definitely the test fixtures. The colors that were selected as defaults by @nishant-karajgikar are not the same as what was used previously. If we wish to fix these tests now then we could regenerate new fixtures. Alternatively we could adjust the default colors chosen in DayStyle.swift so that they match what was used in the past. @1ec5 - Any preference? However if we go the route of making new fixtures then I'll need help since I'm not sure how to regenerate the test fixtures myself.

**Pull request merged**